rtmr/extend.go

```diff
@@ -54,19 +95,47 @@ func ExtendEventLogClient(client configfsi.Client, rtmrIndex int, hashAlgo crypt
 }
 
 // ExtendEventLog extends the measurement into the rtmr with the given hash algorithm and event log.
+// It checks for the existence of the legacy configfs path to decide whether to use the client or sysfs method.
 func ExtendEventLog(rtmrIndex int, hashAlgo crypto.Hash, eventLog []byte) error {
-	client, err := linuxtsm.MakeClient()
-	if err != nil {
-		return err
+	// Defines the legacy rtmr path.
+	const legacyRTMRPath = "/sys/kernel/config/tsm/rtmrs"
+
+	// Check if the legacy configfs path exists to determine the extension method.
+	if _, err := os.Stat(legacyRTMRPath); err == nil {
+		// If the path exists, use the client method.
+		client, err := linuxtsm.MakeClient()
+		if err != nil {
+			return err
+		}
+		return ExtendEventLogClient(client, rtmrIndex, hashAlgo, eventLog)
+	} else if os.IsNotExist(err) {
+		// If the configfs path does not exist, use the sysfs interface instead.
+		return ExtendEventLogSysfs(rtmrIndex, hashAlgo, eventLog)
+	} else {
+		// Handle other unexpected errors from os.Stat.
+		return fmt.Errorf("failed to check for directory %s: %w", legacyRTMRPath, err)
 	}
-	return ExtendEventLogClient(client, rtmrIndex, hashAlgo, eventLog)
 }
 
 // ExtendDigest extends the measurement into the rtmr with the given digest.
+// It checks for the existence of the legacy configfs path to decide whether to use the client or sysfs method.
 func ExtendDigest(rtmrIndex int, digest []byte) error {
-	client, err := linuxtsm.MakeClient()
-	if err != nil {
-		return err
+	// Defines the legacy rtmr path.
+	const legacyRTMRPath = "/sys/kernel/config/tsm/rtmrs"
```

Comment:
this can be a const in this file (so don't need to repeat it in the two functions)